circleci/integration-test

```diff
@@ -73,6 +79,34 @@ fi
 
 echo "Job ID: $JOB_ID"
 
+install_awscli
+# aws login.
+AWS_S3_PROFILE=oidc-s3-profile
+if [[ -v OIDC_S3_UPLOAD_ROLE ]]; then
+  echo "Logging into AWS using role credentials...."
+  assume_role_with_web_identity $OIDC_S3_UPLOAD_ROLE $AWS_S3_PROFILE 
+fi
+
+download_test_artifacts(){
+  echo "Downloading test artifacts"
+  mkdir -p $CIRCLE_ARTIFACTS $CIRCLE_TEST_REPORTS
+  for workflowID in $(cat integration-tests.out | jq -r '.testWorkflowIds[]');
+  do
+    S3_BUCKET_URL="s3://frontend-to-end-tester-files-dev/${workflowID}/artifacts.zip"
+    DEST="${CIRCLE_ARTIFACTS}/${workflowID}/artifacts.zip"
+    echo "  Source: $S3_BUCKET_URL"
+    echo "  Desination: $DEST"
+    if [[ -v OIDC_S3_UPLOAD_ROLE ]]; then
+        echo "Downloading files from S3 using profile ${AWS_S3_PROFILE}" 
+        aws s3 cp --profile $AWS_S3_PROFILE  $S3_BUCKET_URL $DEST
+    else
```

Comment:
is there a case in our ci pipeline where static creds are used? or is this more for simplicity of development when doing this locally?

Reply:
This same logic is used in the `s3-upload` and `catapult-publish` commands, I think as a fallback in case the CI workflow that uses those commands doesn't use the CircleCI context `aws-s3-upload`. 
Do you think we should just remove this and make some PRs in all workflow jobs that use this script to use the `aws-s3-upload` context?

Reply:
i think so personally, but i don't feel too strongly about it either way

Reply:
@ulziibay do you have any thoughts about this? Is there something I'm missing about the `OIDC_S3_UPLOAD_ROLE` and `aws-s3-upload` context / is there a case where our CI pipeline isn't able to use the CircleCI context?

Reply:
All of our CI pipelines will have access to `aws-s3-upload` CI context. However, each workflow must declare "context: aws-s3-upload" in order for the jobs in that workflow to have access to the env variables defined in the workflow. 

In general, to utilize `aws-s3-upload` context correctly, you must enable `context: aws-s3-upload` to all the places in which `integration-test` is invoked. That could be quite challenging to to at scale. Therefore, checking if `OIDC_S3_UPLOAD_ROLE` is defined is a fail-safe way to make sure your S3 upload succeeds either with static credentials or via the role based temporary credentials. 

In addition, to utilize `aws-s3-upload` , the role based policy must perhaps be changed  ✅ , which seems to have been done in https://github.com/Clever/infra/pull/2689